plugins/modules/dcnm_image_policy.py

```diff
@@ -0,0 +1,694 @@
+
+        # Exit if there's nothing to do
+        if len(self.want) == 0:
+            self.ansible_module.exit_json(**self.task_result.module_result)
+
+    def handle_replaced_state(self) -> None:
+        """
+        Replace all policies on the controller that are in want
+        """
+        instance = ImagePolicyReplaceBulk(self.ansible_module)
+        instance.payloads = self.want
+        instance.commit()
+        self.update_diff_and_response(instance)
+
+    def handle_deleted_state(self) -> None:
+        """
+        1.  Delete all policies in self.want that exist on the controller
+        """
+        instance = ImagePolicyDelete(self.ansible_module)
+        policy_names_to_delete = []
+        for want in self.want:
+            if want["policyName"] in self.have.all_policies:
+                policy_names_to_delete.append(want["policyName"])
+
+        msg = f"policy_names_to_delete: {policy_names_to_delete}, "
+        msg += f"type {type(policy_names_to_delete).__name__}"
+        self.log.debug(msg)
+        instance.policy_names = policy_names_to_delete
+        instance.commit()
+        self.update_diff_and_response(instance)
+
+    def _delete_policies_not_in_want(self) -> None:
+        """
+        Delete all policies on the controller that are not in self.want
+
+        Caller: handle_overridden_state()
+        """
+        want_policy_names = set()
+        for want in self.want:
+            want_policy_names.add(want["policyName"])
+
+        policy_names_to_delete = []
+        for policy_name in self.have.all_policies:
+            if policy_name not in want_policy_names:
+                policy_names_to_delete.append(policy_name)
+
+        instance = ImagePolicyDelete(self.ansible_module)
+        instance.policy_names = policy_names_to_delete
+        instance.commit()
+        self.update_diff_and_response(instance)
+
+    def handle_query_state(self) -> None:
+        """
+        1.  query the policies in self.want that exist on the controller
+        """
+        instance = ImagePolicyQuery(self.ansible_module)
+        policy_names_to_query = []
+        for want in self.want:
+            policy_names_to_query.append(want["policyName"])
+        instance.policy_names = policy_names_to_query
+        instance.commit()
+        self.update_diff_and_response(instance)
+
+    def handle_overridden_state(self) -> None:
+        """
+        1.  Delete all policies on the controller that are not in self.want
+        2.  Call handle_merged_state()
+        """
+        self._delete_policies_not_in_want()
+        self.handle_merged_state()
+
+    def handle_merged_state(self) -> None:
+        """
+        1.  Populate self.need_create with items from self.want that are
+            not in self.have
+        2.  Populate self.need_update with updated policies.  We update
+            policies as follows:
+            a.  If a policy is in both self.want amd self.have, and they
```

Comment:
So based on this it looks like merge will update changed properties.  From the description in the doc header at the beginning this was not clear to me but I could have misunderstood

Reply:
Good catch.  Yes, merge will:

1. Create policies in self.want that do not exist on the controller
2. Update policies in self.want that DO exist on the controller

I've updated the doc and will commit the change in a sec...